### PR TITLE
Use table role in calendar

### PR DESCRIPTION
**Pull request by YueyingLu**
### Description

AWSUI-19185
Removing role "none" and "button" from the table. 

### How has this been tested?

[_How did you test to verify your changes?_]

[_How can reviewers test these changes efficiently?_]

[_Check for unexpected visual regressions, see [`CONTRIBUTING.md`](CONTRIBUTING.md#run-visual-regression-tests) for details._]

### Documentation changes

[*Do the changes include any API documentation changes?*]
- [ ] _Yes, this change contains documentation changes._
- [ ] _No._

### Related Links

[*Attach any related links/pull request for this change*]

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- [ ] _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](CONTRIBUTING.md#public-apis)._
- [ ] _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](CONTRIBUTING.md#browsers-support)._
- [ ] _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- [ ] _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- [ ] _Changes are covered with new/existing unit tests?_
- [ ] _Changes are covered with new/existing integration tests?_
</details>

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/calendar/grid/index.tsx [hunk 1]
```diff
@@ -14,6 +14,15 @@ import useFocusVisible from '../../internal/hooks/focus-visible/index.js';
 import clsx from 'clsx';
 import { useEffectOnUpdate } from '../../internal/hooks/use-effect-on-update.js';
 
+/**
+ * There are some known issues with screenreaders, they are reproducible on https://www.w3.org/WAI/ARIA/apg/example-index/dialog-modal/datepicker-dialog
```
Comment (pan-kot) [inline src/calendar/grid/index.tsx:18]: style: "There are known issues with browser/screenreader combinations also reproducible on ..."

File: src/calendar/grid/index.tsx [hunk 2]
```diff
@@ -111,7 +120,8 @@ export default function Grid({
               scope="col"
               className={clsx(styles['calendar-grid-cell'], styles['calendar-day-header'])}
             >
-              {renderDayName(locale, dayIndex)}
+              <span aria-hidden="true">{renderDayName(locale, dayIndex, 'short')}</span>
+              <span className={styles['visually-hidden']}>{renderDayName(locale, dayIndex, 'long')}</span>
```
Comment (pan-kot) [inline src/calendar/grid/index.tsx:124]: How about changing "visually-hidden" to "screenreader-only"? I think it makes the code better readable by answering the question "why?" instead of "what?".

Comment (YueyingLu) [reply src/calendar/grid/index.tsx:124]: We have "visually-hidden" in Formfield too. Let's refactor it as a whole as followup of this ticket.

**Comment by pan-kot** [comment]
Testing:

1. Safari/Chrome+VO - when using application navigation the announcements are not assertive and if navigating fast some dates will be skipped.
2. NVDA next table command doesn't find the calendar ("no next table") - I am not sure if it should since we use role="grid".
3. Chrome+NVDA seems to not use table navigation. Using Control+Alt+Arrow-keys just moves the application focus.

File: src/date-picker/__tests__/date-picker-calendar.test.tsx [hunk 3]
```diff
@@ -435,7 +437,10 @@ describe('Date picker calendar', () => {
         const { wrapper } = renderDatePicker({ ...defaultProps, value: '2018-03-21', isDateEnabled });
         act(() => wrapper.findOpenCalendarButton().click());
         expect(findCalendarHeaderText(wrapper)).toBe('March 2018');
-        expect(wrapper.findCalendar()!.findSelectedDate().getElement().textContent).toBe('21');
+        expect(
+          wrapper.findCalendar()!.findSelectedDate()?.find(`:not(.${calendarStyles['visually-hidden']})`)?.getElement()
```
Comment (pan-kot) [inline src/date-picker/__tests__/date-picker-calendar.test.tsx:441]: The `findSelectedDate` is a public selector. We need to exclude the hidden content as part of it to not break the customer code.

**Review by pan-kot — CHANGES_REQUESTED** [review]

File: src/calendar/grid/index.tsx [hunk 4]
```diff
@@ -151,10 +168,9 @@ export default function Grid({
...
                   key={`${weekIndex}:${dateIndex}`}
                   ref={tabIndex === 0 ? focusedDateRef : undefined}
-                  role="button"
                   tabIndex={tabIndex}
-                  aria-label={dayAnnouncement}
-                  aria-current={isSelected ? 'date' : undefined}
+                  aria-current={isDateOnSameDay ? 'date' : undefined}
+                  aria-selected={isSelected}
                   aria-disabled={!isEnabled}
                   onClick={onClick}
```
Comment (pan-kot) [inline src/calendar/grid/index.tsx:175]: In Safari+VO the "clickable" announcement is also done for disabled days when using table navigation.
Probably need to attach the onClick handler conditionally like `{ isEnabled ? () => onSelectDate(date) : undefined }` and add a test to ensure the handler is not added for disabled days.

File: src/calendar/grid/index.tsx [hunk 5]
```diff
@@ -14,6 +14,22 @@ import useFocusVisible from '../../internal/hooks/focus-visible/index.js';
...
+ * 2. Safari+VO — “dimmed” day state is announced twice
+ * 3. NVDA+Firefox - can not move focus by arrow up/down
+ * 4. Chrome+NVDA - seems to not use table navigation. Using Control+Alt+Arrow-keys just moves the application focus
+ * Alternatively, users can access calendar with keyboard application navigation.
+ * Following are known issues with keyboard application navigation which are reproducible on https://www.w3.org/WAI/ARIA/apg/example-index/dialog-modal/datepicker-dialog :
+ * 1. NVDA+Firefox — every day is announced as “not selected”
+ * 2. VO - doesn't announce column header when focusing on days
+ * 3. VO - no indication that the days are clickable(not selected, clickable...)
+ * 4. VO - announcements are not assertive and if navigating fast some dates will be skipped
+ */
```
Comment (pan-kot) [inline src/calendar/grid/index.tsx:31]: I'd structure the message a little different (+ small updates to the bug descriptions):

```
Calendar grid supports two mechanisms of keyboard navigation:
- Native screen-reader table navigation (semantic table markup);
- Keyboard arrow-keys navigation (a custom key-down handler).

The implementation largely follows the w3 example (https://www.w3.org/WAI/ARIA/apg/example-index/dialog-modal/datepicker-dialog) and shares the following issues:
- (table navigation) Chrome+VO - weekday is announced twice when navigating to the calendar's header;
- (table navigation) Safari+VO - "dimmed" state is announced twice;
- (table navigation) Firefox/Chrome+NVDA - cannot use table navigation if any cell has a focus;
- (keyboard navigation) Firefox+NVDA - every day is announced as "not selected";
- (keyboard navigation) Safari/Chrome+VO - weekdays are not announced;
- (keyboard navigation) Safari/Chrome+VO - days are not announced as interactive (clickable or selectable);
- (keyboard navigation) Safari/Chrome+VO - day announcements are not interruptive and can be missed if navigating fast.
```

What do you think?

File: src/date-picker/__integ__/page-objects/date-picker-page.ts [hunk 6]
```diff
@@ -106,11 +107,11 @@ export default class DatePickerPage extends BasePageObject {
   }
 
   isDateFocused() {
-    return this.isFocused(this.findCalendar().findSelectedDate().getElement());
+    return this.isFocused(this.findCalendar().find(`.${calendarStyles['calendar-day-selected']}`).getElement());
```
Comment (pan-kot) [inline src/date-picker/__integ__/page-objects/date-picker-page.ts:110]: why this change is needed?

File: src/date-picker/__tests__/date-picker-calendar.test.tsx [hunk 7]
```diff
@@ -40,7 +40,9 @@ describe('Date picker calendar', () => {
   }
 
   const findFocusedDay = (wrapper: DatePickerWrapper) => {
-    return wrapper.findCalendar()!.find(`.${calendarStyles['calendar-day']}[tabIndex="0"]`);
+    return wrapper
+      .findCalendar()!
+      .find(`.${calendarStyles['calendar-day']}[tabIndex="0"] :not(.${calendarStyles['visually-hidden']})`);
```
Comment (pan-kot) [inline src/date-picker/__tests__/date-picker-calendar.test.tsx:45]: Probably can update this one with:

```
wrapper
  .findCalendar()!
  .findComponent(`.${calendarStyles['calendar-day']}[tabIndex="0"]`, CalendarDayWrapper)!
  .findLabel()
```
That would ensure regression for the new `findLabel` util.

Comment (pan-kot) [reply src/date-picker/__tests__/date-picker-calendar.test.tsx:45]: Noticed that findLabel is already used below - resolving conversation

**Review by pan-kot — DISMISSED** [review]
LGTM!

**Review by pan-kot — DISMISSED** [review]

**Review by pan-kot — DISMISSED** [review]

**Review by pan-kot — APPROVED** [review]
